In [28]:
import pandas as pd
import numpy as np
from scipy import stats as stats
from scipy.stats import norm
from scipy.stats import chi2_contingency
from scipy.stats import chi2

#Q1

In [2]:
from google.colab import files
uploaded=files.upload()

Saving Cutlets.csv to Cutlets.csv


Null hyposthesis as Ho: μ1 = μ2 (There is no difference in diameters of cutlets between two units).

Alternate hypothesis as Ha: µ1 µ2 (There is significant difference in diameters of cutlets between two units).

In [3]:
df=pd.read_csv('Cutlets.csv')

In [4]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [7]:
UnitA=pd.Series(df.iloc[:,0])
UnitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [8]:
UnitB=pd.Series(df.iloc[:,1])
UnitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [9]:
#2-Sample 2-Tail t-test
p_value=stats.ttest_ind(UnitA,UnitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)


we get           
p = 0.4722394724599501           
alpha = 0.05           
  comparing p and alpha.          
So we get         
      p> alpha      
Inference: Accept Null hyposthesis as Ho: u1 μ2, There is no difference in diameters of cutlets between two units.

#Q2

In [10]:
from google.colab import files
uploaded=files.upload()

Saving LabTAT.csv to LabTAT.csv


Anova f-test statistics: Analysis of varaince between more than 2 samples or columns.         
Null Hypothesis Ho as No Varaince: All samples TAT population means are
same.          
Alternate Hypothesis HA as It has Variance: Atleast one sample TAT population mean is different.

In [11]:
df=pd.read_csv('LabTAT.csv')

In [12]:
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [13]:
#Anova f-test statistics of oneway
p=stats.f_oneway(df.iloc[:,0],df.iloc[:,1],df.iloc[:,2],df.iloc[:,3])
p

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

we get          
p = 2.1156708949992414e-57              
alpha = 0.05             
comparing p and alpha            
So we get            
p< alpha                
Inference: Reject Null Hypothesis Ho.                
Thus, Accept Alternate Hypothesis HA as It has Variance: Atleast one sample TAT population mean is different.

#Q3

In [14]:
from google.colab import files
uploaded=files.upload()

Saving BuyerRatio.csv to BuyerRatio.csv


Null Hypothesis as HO: All proportions are equal.            
Alternative hypothesis as Ha: Not all Proportions are equal.

In [15]:
df=pd.read_csv('BuyerRatio.csv')

In [16]:
df.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [18]:
df_table=df.iloc[:,1:6]
df_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [19]:
df_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [21]:
val=stats.chi2_contingency(df_table)
val

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [22]:
type(val)

tuple

In [24]:
no_of_rows=len(df_table.iloc[0:2,0])
no_of_columns=len(df_table.iloc[0,0:4])
degree_of_f=(no_of_rows-1)*(no_of_columns-1)
print('DOF=',degree_of_f)

DOF= 3


In [25]:
Expected_value=val[3]
Expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [29]:
chi_square=sum([(o-e)**2/e for o,e in zip(df_table.values,Expected_value)])
chi_square_statestic=chi_square[0]+chi_square[1]
chi_square_statestic

1.5152956451130446

In [27]:
critical_value=chi2.ppf(0.95,3)
critical_valuecritical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

In [31]:
pvalue=1-chi2.cdf(chi_square_statestic,3)
pvalue

0.6787446296467897

P=0.6787446296467897                    
Alpha=0.05                  
comparing p and alpha                       
So we get             
p> alpha                           
Inference: Accept Null Hypothesis H0               
So, all proportions are equal.

#Q4

In [33]:
from google.colab import files
uploaded=files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


Null Hypothesis as Ho: Independence of categorical variables (customer order forms defective % does not varies by centre).            
Alternative hypothesis as Ha: Dependence of categorical variables (customer order forms defective % varies by centre).

In [34]:
df=pd.read_csv('Costomer+OrderForm.csv')

In [35]:
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [36]:
df.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [37]:
df.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [38]:
df.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [39]:
df.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [40]:
datap=np.array([[271,267,269,280],[29,33,31,20]])
datap

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [41]:
chi2_contingency(datap)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

we get            
p = 0.2771020991233135                 
alpha = 0.05                
comparing p and alpha                
So we get               
p> alpha                 
Inference: Accept Null Hypthesis HO i.e. Independence of categorical variables.                
Thus, customer order forms defective % does not varies by centre.